aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -53,17 +53,26 @@ static inline Tensor unary_op_impl(const Tensor& self, OutImpl& out_impl) {
   return out_impl(result, self);
 }
 
+// Implicitly promote all int and bool dtypes to default floating dtype
+// See PR #33322 for more discussion on this
+template <typename OutImpl>
+static inline Tensor unary_float_op_impl(const Tensor& self, OutImpl& out_impl) {
+  Tensor result = isIntegralType(self.scalar_type(), /*include_bool=*/ true) ? at::empty({0}, self.options().dtype(typeMetaToScalarType(c10::get_default_dtype()))) : \
```

Comment:
The linter will likely complain about line length. You can break this line at the "?".

Reply:
I agree, I should have done this at the first place. Thanks!

Reply:
semi-related tip:  I know you like vim.  `/ $` is helpful to highlight/catch trailing whitespace.

Reply:
Thanks @mcarilli. :+1:

Reply:
Removed white-space and added break line! (https://github.com/pytorch/pytorch/pull/33322/commits/c816f39f45369536c2ffb329756ae36405b21d64)